model/mobile_trigger.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type MobileTrigger struct {
+	PluginId string `json:"id"`
+	Location string `json:"location"`
+	Trigger  string `json:"trigger"`
+	//	Extra    interface{} `json:"extra"`
+	Extra string `json:"extra"` // Should use interface{} to be able to create different extras for different locations
+}
```

Comment:
I would like to see a `Request URL` field here. It would allow plugins (and potential other integrations) to specify there API themself and don't rely on a pre-defined pattern. It also follows what we have been doing with e.g. post actions.